enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -107,51 +103,134 @@ users' workload Pods so that containers in those Pods can install RHEL subscript
 
 ### Implementation Details/Notes/Constraints [optional]
 
 #### Install
 
-A comparison of installing the implementation of this proposal via OLM vs. CVO was done in another enhancement proposal.  
-The net result was to go with CVO.
+As per the open questions, whether to go with a CVO or OLM based approach, or perhaps some combination, is a question
+with no current consensus on the answer.
 
 Some specifics:
 - During the review for the ["Installation of CSI drivers that replace in-tree drivers](https://github.com/openshift/enhancements/blob/master/enhancements/storage/csi-driver-install.md)
-a question was posed about whether the exact type of CSI driver this proposal is describing would use the method 
+a question posed about whether the exact type of CSI driver this proposal is describing would use the method 
 described in that document (namely OLM).  The recommendation from the [storage SMEs was to go with CVO](https://github.com/openshift/enhancements/pull/139/files#r391761362)
-- In particular, using CVO and having it be a day 1 install operation meets the ease of use targets we are trying to
+was to not use OLM, and use CVO instead.
+- Using CVO and thus making it a day 1 install operation meets the ease of use targets we are trying to
 reach with this proposal
 - Crafting OLM dependencies between this proposal and [related proposals](https://github.com/openshift/enhancements/pull/214)
-as part of an OLM based install at the time of this writing is immature.
+as part of an OLM based install at the time of this writing is immature, at least for some reviewers.
+- However, other reviewers wonder if we are in fact close from having either OLM or CVO operators being able to require
+an OLM operator to be present.
+- Using OLM avoids further increasing the install payload size.
+- Using OLM avoids unexpected hiccups in the feature's installation impacting initial cluster install or cluster upgrade.
+
+#### End to end flow (from creation, to opt-in, to validation, and then injections)
+
+An admin creates a new cluster level custom resource for encapsulating the sharable `Secret` or `ConfigMap`.
+
+
+```yaml
+type: ProjectedResource
+name: SharedExternalAccessSecret
+spec:
+  backingResource:
+    type: Secret
+    namespace: ns-one
+    name: cool-secret
+  description: "used to access my internal git repo, RHEL entitlements, or something else cool"
+status:
+  conditions: available
+```
+
+And sets up ACL/RBAC to control who can list/get such objects.
+
+A user queries this new type to see what is available to her/him.
+
+Then, to opt in, a user created Pod can cite a CSI volume references (see below for a more detailed explanation of them) that 
+references our new type:
+
+```yaml
+csi:
+  driver: projectedresource.storage.openshift.io
+    volumeAttributes:
+      projectedResourceName: the-projected-resource
+
+```
+
+For pre-existing objects that encapsulate a Pod, a new, documented annotation can be declared to facilitate injection
```

Comment:
if we expose names in a spot users can read, we may not need to have an annotation that does this.  The user can simply list the ProjectedResource in their pod because the name will be presumed stable and discoverable in a way the the previous: specify-a-namespace-and-name was not.

Reply:
I'm missing something here @deads2k  ... *how* exactly does the user list the `ProjectedResource` in their pod?  Or another way, what yaml snippet are you referring to when you say this?

thanks

Reply:
wouldn't they be listing it in their pod volume definitions?

Reply:
I agree we don't need an annotation for the CSI driver. For entitlements we will need to build on top of this:

1. An operator that does the following:
  a. Creates the `ProjectedResource` for the entitlement `Secrets` and `ConfigMaps` that live in `openshift-config`
  b. Creates RBAC rules for the entitlements.
  c. Adds appropriate `ClusterRoleBindings` (perhaps based on user config - will address in a future EP).
2. An admission webhook which does the following:
  a. Looks for the entitlement annotation on all created pods
  b. Adds the CSI volume mount(s) for the entitlement `ProjectedResources`, and adds appropriate `volumeMount` options for all containers.

Reply:
> wouldn't they be listing it in their pod volume definitions?

Yes, perhaps, @bparees 

I'm looking to confirm that interpretation is what @deads2k was alluding to.

Reply:
> I agree we don't need an annotation for the CSI driver. For entitlements we will need to build on top of this:
> 
>     1. An operator that does the following:
>        a. Creates the `ProjectedResource` for the entitlement `Secrets` and `ConfigMaps` that live in `openshift-config`
>        b. Creates RBAC rules for the entitlements.
>        c. Adds appropriate `ClusterRoleBindings` (perhaps based on user config - will address in a future EP).
> 
>     2. An admission webhook which does the following:
>        a. Looks for the entitlement annotation on all created pods

The annotation I'm referring to is this one @adambkaplan.  Per prior discussions I've moved the line a bit on which bits we discuss here vs. in say an admission plugin proposal.  It was becoming clear from prior reviews of this proposal was that some sort of high level end of end flow was needed in this doc.

That said, per earlier comments on this thread, and making sure I fully understand @deads2k comment, I may either remove this line, or much further clarify that it is a nice to have.  Probably will do the former and remove, but am not pulling the trigger just yet.

>        b. Adds the CSI volume mount(s) for the entitlement `ProjectedResources`, and adds appropriate `volumeMount` options for all containers.

Reply:
So @deads2k and I talked ... basically we are on the same page.  Nothing blocking what is currently here, but I'm going to add some clarifying text in my next push to cover the various bases touched on this thread.

Reply:
ok @bparees @deads2k @adambkaplan I've pushed an update to the end to end flow with those clarifying updates I referenced in https://github.com/openshift/enhancements/pull/250#discussion_r409133767 to touch about "to annotate or not to annotate" as well as some minor additional details here and there